libp2p/types/libp2p.go

```diff
@@ -0,0 +1,33 @@
+//go:build !production
+
+package types
+
+//go:generate mockgen -source=$GOFILE -destination=./mocks/libp2p_mock.go
+
+import (
+	"context"
+
+	"github.com/libp2p/go-libp2p/core/host"
+)
+
+type Host interface {
+	host.Host
+}
+
+//type Message interface {
+//	// TODO: what?
+//}
+
+type Subscription interface {
+	Topic() string
+	Next(ctx context.Context) (interface{}, error)
+}
+
+//type SubOpt interface {
+//	// TODO: what?
+//}
+
+type Topic interface {
```

Comment:
A small sentence above each interface would really help.

Topic: LibP2P listeners can subscribe on topics, given  a set of options and its name
Subscription: A stream from which messages can be read.

Reply:
I added comments to the place where we're using these constructs from libp2p. This was part of bit of a misadventure of mine that I unintentionally committed.